isarengineering/SecPi
# Several improvements and features

## Summary
- Get rid of using the filesystem completely
- Rework action (response) handling thoroughly
- Use TOML for configuration files
- More [OOP](https://en.wikipedia.org/wiki/Object-oriented_programming)

## Details

### Machinery improvements
- Add `dataclass` objects `AlarmMessage`, `NotificationMessage`, `ActionRequestMessage`
- Use data model to forward responses from Actions, instead of using the filesystem
- Manager: Modernize action machinery (6b6fbcf77717)
- Notifier: Modernize `notify` interface (7da4cb5aca6)
- Configuration: Switch from JSON to TOML

**Note: Two configuration variables have changed:**
- `holddown_timer` is now called `holddown_time`
- `data_timeout` is now called `action_response_timeout`

### Miscellaneous
- Worker: Improve queue handling for initial configuration request
- Webinterface: Rename `secpi.web.sites` to `secpi.web.page`
- Webinterface: Improve bootstrapping
- Improve documentation and logging
- Refactoring and naming things
- ADAM-6000: Add reading from digital outputs
- Improve application bootstrapping
- Add options `write_protected` and `config_no_update` to worker configuration
- Use URI-style configuration for connecting to AMQP broker
- Naming things: Replace ``pi_id`` with ``worker_id`` across the board

### Bugfixes
- FFMPEGVideo: Fix return value
- AMQP: Fix queue declarations and bindings
- Actions: Fix propagating and evaluating `active_state` attribute
- Mailer: Fix hiccups with charset encoding

### QA
- Add `ruff` linter
- Add a "full stack" test case
- Add a test case to verify that all configuration files in the repository are syntactically valid
